docs/pages/architecture/authorization.mdx

```diff
@@ -0,0 +1,386 @@
+
+#### External users from SSO providers
+
+Let's review the example Alice who is authenticating using SSO provider of her organization
+with Teleport:
+
+![Role mapping](../../img/architecture/role-mapping@1.5x.svg)
+
+<Notice type="tip">
+Every time SSO user logs in, Teleport creates a temporary user account record
+that automatically expires with SSO session and logs audit log entries.
+
+Teleport creates this record to avoid name collisions with local users.
+</Notice>
+
+#### External users from other clusters
+
+A user could be external to teleport cluster, if another cluster or certificate authority issues a certificate
+that this cluster trusts. In this case, Teleport activates [trusted cluster mapping logic](./trustedclusters.mdx#role-mapping).
+
+#### Local interactive users
+
+Local interactive users have a record in Teleport's backend with credentials. Their user entry is assign
+
+A cluster administrator have to create account entries for every Teleport user with
+[`tctl users add`](../setup/reference/cli.mdx) or API call.
+
+Every local Teleport User must be associated with a list of one or more roles.
+This list is called "role mappings".
+
+### Non-interactive users
+
+Teleport supports non-interactive users for automation services, e.g. Jenkins or micro-services
+running in your organization.
+
+### Local non-interactive users
+
+Local non-interactive users also have a user entry that maps their name to roles,
+but they do not have credentials stored in the database.
+Non-interactive users have to use Teleport's machine ID product to receive and renew certificates.
+Teleport Machine ID's bot runs alongside services and rotates SSH and X.509 certificates on behalf
+of non-interactive users:
+
+<Figure
+  align="center"
+  bordered="true"
+  caption="Machine-ID certs"
+  >
+  
+![Certificates for services](../../img/architecture/certs-machine-id@1.8x.svg)
+</Figure>
+
+### External non-interactive users
+
+External non-interactive users behave just like local ones, but it is another
+cluster or certificate authority that issues certificates for them.
+
+They do not have local user records in Teleport backend. Teleport activates
+[trusted cluster mapping logic](./trustedclusters.mdx#role-mapping) to support this use case.
+
+## Role Based Access Control
+
+Every Teleport user is assigned one or several roles that govern access to resources and Teleport's API.
+
+### Allow and Deny Rules
+
+Each Teleport role works by having two lists of rules: `allow` rules and `deny` rules:
+
+- Everything is denied by default.
+- Deny rules get evaluated first and take priority.
+- A rule consists of two parts: the resources and verbs.
+
+Here's an example of an allow rule describing a list verb applied to the sessions resource.
+It means "allow users of this role to see a list of recorded SSH or Kubernetes sessions".
+
+```yaml
+allow:
+  - resources: [session]
+    verbs: [list]
```

Comment:
We ought to have a full list of all the verbs that we support somewhere and a description of exactly what they allow/disallow. Same with all the resource types - preferably something generated automatically from source!

Reply:
should be in the reference already